tests/test_ag3.py

```diff
@@ -1026,44 +1036,57 @@ def test_gene_cnv_frequencies(contig):
         "bf_2012_col_amp",
         "bf_2012_col_del",
     ]
-    df_genes = ag3.geneset().query(f"type == 'gene' and contig == '{contig}'")
 
-    df = ag3.gene_cnv_frequencies(contig=contig, sample_sets="v3_wild", cohorts=cohorts)
-
-    assert isinstance(df, pd.DataFrame)
-    assert df.columns.tolist() == expected_cols
-    assert len(df) == len(df_genes)
-    assert df.index.name == "ID"
-
-    # sanity checks
-    for f in ["ke_amp", "ke_del", "bf_2012_col_amp", "bf_2012_col_del"]:
-        x = df[f].values
-        assert np.all(x >= 0)
-        assert np.all(x <= 1)
-    for fa, fd in [["ke_amp", "ke_del"], ["bf_2012_col_amp", "bf_2012_col_del"]]:
-        a = df[fa].values
-        d = df[fd].values
-        x = a + d
-        assert np.all(x >= 0)
-        assert np.all(x <= 1)
-
-
-@pytest.mark.parametrize(
-    "contig",
-    [
-        "X",
-    ],
-)
-def test_gene_cnv_frequencies_0_cohort(contig):
     ag3 = setup_ag3()
-    cohorts = {
-        "bf_2050_col": "country == 'Burkina Faso' and year == 2050 and species == 'coluzzii'",
-    }
-    with pytest.raises(ValueError):
-        _ = ag3.gene_cnv_frequencies(
+    df_genes = ag3.geneset().query(f"type == 'gene' and contig == '{contig}'")
+    if "bf_2050_col" in cohorts:
+        with pytest.raises(ValueError):
+            _ = ag3.gene_cnv_frequencies(
+                contig=contig, sample_sets="v3_wild", cohorts=cohorts
+            )
+    else:
+        df = ag3.gene_cnv_frequencies(
             contig=contig, sample_sets="v3_wild", cohorts=cohorts
         )
 
+        assert isinstance(df, pd.DataFrame)
+        assert len(df) == len(df_genes)
+        assert df.index.name == "ID"
+
+        # sanity checks
+        if isinstance(cohorts, dict):
+            for f in ["ke_amp", "ke_del", "bf_2012_col_amp", "bf_2012_col_del"]:
+                x = df[f].values
+                assert np.all(x >= 0)
+                assert np.all(x <= 1)
+            for fa, fd in [
+                ["ke_amp", "ke_del"],
+                ["bf_2012_col_amp", "bf_2012_col_del"],
+            ]:
+                a = df[fa].values
+                d = df[fd].values
+                x = a + d
+                assert np.all(x >= 0)
+                assert np.all(x <= 1)
+                assert df.columns.tolist() == expected_fields
+                if contig == "2R":
+                    assert df.shape == (3668, 10)
+                elif contig == "X":
+                    assert df.shape == (1063, 10)
```

Comment:
This can be removed, because it is already covered by better cross-checks above. I.e., the length of the dataframe is checked by comparing with the length of `df_genes`, and the columns are checked by comparing to `expected_fields`.